webclient/components/1time-Mrow-mField/diagnosis/edit-layer/add-ct.vue

```diff
@@ -0,0 +1,297 @@
+      v-if="cfGetClientSideTableApiSendingStateRows.length > 0"
+      :data="cfGetClientSideTableApiSendingStateRows"
+      style="width: 100%; background: #f0f9eb">
+      <el-table-column label="Diagnosiss sending to server">
+        <div slot-scope="{row}" class="td-number">
+          {{fnGetDiagnosisNameById(row.masterDiagnosisId)}}
+        </div>
+      </el-table-column>
+      <!-- <el-table-column prop="diagnosisName" label="Diagnosiss sending to server"></el-table-column> -->
+    </el-table>
+
+    <!-- Goal: Show data saved successfuly this session -->
+    <el-table
+      v-if="cfGetClientSideTableApiSuccessStateRows.length > 0"
+      :data="cfGetClientSideTableApiSuccessStateRows"
+      style="width: 100%; background: #f0f9eb">
+      <el-table-column label="Diagnosiss added this session">
+        <div slot-scope="{row}" class="td-number">
+          {{fnGetDiagnosisNameById(row.masterDiagnosisId)}}
+        </div>
+      </el-table-column>
+    </el-table>
+    <!-- Goal: Show data of API that failed in this session -->
+    <el-table
+      v-if="cfGetClientSideTableApiErrorStateRows.length > 0"
+      :data="cfGetClientSideTableApiErrorStateRows"
+      style="width: 100%; background: #f0f9eb"
+    >
+    <el-table-column label="Error: Diagnosiss attempted but failed to save">
+      <div slot-scope="{row}" class="td-number">
+        {{fnGetDiagnosisNameById(row.masterDiagnosisId)}}
+      </div>
+    </el-table-column>
+    </el-table>
+  </div>
+</template>
+<script>
+
+import clientSideTblPatientDiagnosis from '../db/client-side/structure/patient-table-of-diagnosis'
+import clientSideTblMasterDiagnosis from '../db/client-side/structure/master-table-of-diagnosis'
+
+export default {
+  data() {
+    return {
+      isClickReviewed: false
+    }
+  },
+  computed: {
+    // clientSideTblPatientDiagnosis functions can not be directly called from template. hence computed functions have been defined.
+    cfGetClientSideTableNewRowsInEditState() {
+      return clientSideTblPatientDiagnosis.fnGetNewRowsInEditState()
+    },
+    cfGetClientSideTableReadyToReviewedStateRows() {
+      return clientSideTblPatientDiagnosis.fnGetNewRowsInReadyToReviewedState()
+    },
+    cfGetClientSideTableApiSuccessStateRows() {
+      return clientSideTblPatientDiagnosis.fnGetNewRowsInApiSuccessState()
+    },
+    cfGetClientSideTableApiErrorStateRows() {
+      return clientSideTblPatientDiagnosis.fnGetNewRowsInApiErrorState()
+    },
+    cfGetClientSideTableApiSendingStateRows() {
+      return clientSideTblPatientDiagnosis.fnGetNewRowsInApiSendingState();
+    },
+  },
+  methods: {
+    fnGetDiagnosisNameById(masterDiagnosisId) {
+      let getMasterOfDiagnosis = clientSideTblMasterDiagnosis
+        .query()
+        .where('masterDiagnosisId', masterDiagnosisId)
+        .first()
+        
+      return getMasterOfDiagnosis != null ? getMasterOfDiagnosis.diagnosisName : '';
+    },
+    cfGetMasterListOfDiagnosis: function () {
```

Comment:
This should be computed function

Reply:
Rename the function

Reply:
This function not needed to computed function